### ISSUE TITLE
[ML] Implement DBSCAN Ensemble Anomaly Detection with Civic-Domain Presets

### ISSUE THREAD

**Issue by POWDER-RANGER** [POWDER-RANGER/CIVWATCH#182]
## Objective

Implement production-ready anomaly detection using a multi-algorithm ensemble (DBSCAN + Isolation Forest + Z-Score) with civic-domain-specific tuning presets. This replaces the current rule-based anomaly flagging with ML-powered detection.

## Background

Currently, CIVWATCH has basic threshold-based alerting. This milestone implements the full anomaly detection pipeline described in [ML_TUNING.md](https://github.com/POWDER-RANGER/CIVWATCH/blob/main/ML_TUNING.md), including the DBSCAN-based clustering approach from Ester et al. (1996) combined with ensemble voting for robust detection across diverse civic data types.

### 1. DBSCAN Implementation
- [ ] Core DBSCAN clustering with configurable `eps`, `min_samples`, `metric`
- [ ] K-distance graph utility for `eps` parameter selection
- [ ] RobustScaler preprocessing for financial data resilience
- [ ] Feature vector construction (numerical, temporal, categorical, textual)
- [ ] Noise point identification and labeling

**Reference**: [ML_TUNING.md — DBSCAN Parameters](https://github.com/POWDER-RANGER/CIVWATCH/blob/main/ML_TUNING.md#dbscan-parameters)  
**Paper**: [Ester et al., KDD-96](https://www.dbs.ifi.lmu.de/Publikationen/Papers/KDD-96.final.frame.pdf)

### 2. Ensemble Voting System
- [ ] Isolation Forest integration (`sklearn.ensemble.IsolationForest`)
- [ ] Statistical Z-Score detector (with MAD for robustness)
- [ ] Weighted ensemble voting (DBSCAN: 0.4, IF: 0.35, Z-Score: 0.25)
- [ ] Configurable weights per monitor/domain
- [ ] Confidence score calibration

```python
# Target ensemble configuration
ANOMALY_DETECTORS = {
    "dbscan": {"enabled": True, "weight": 0.4, "params": {...}},
    "isolation_forest": {"enabled": True, "weight": 0.35, "params": {...}},
    "z_score": {"enabled": True, "weight": 0.25, "params": {...}}
}
```

### 3. Civic-Domain Presets
- [ ] **Campaign Finance**: `eps=0.3, min_samples=8, contamination=0.03`
- [ ] **Meeting Minutes**: `eps=0.7, min_samples=5, contamination=0.08`
- [ ] **Contract Awards**: `eps=0.4, min_samples=10, contamination=0.02`
- [ ] **FOIA Requests**: `eps=0.6, min_samples=4, contamination=0.05`
- [ ] **311 Service**: `eps=0.5, min_samples=6, contamination=0.06`
- [ ] Auto-detection of optimal preset based on source type

### 4. Feature Engineering Pipeline
- [ ] Numerical: dollar amounts, counts, sentiment scores
- [ ] Temporal: hour, day_of_week, month, days_since_activity
- [ ] Categorical: source type, entity type, region (one-hot encoded)
- [ ] Textual: TF-IDF (top 500), topic distribution, key phrase presence
- [ ] Feature Store integration (Feast) for online/offline consistency

### 5. Model Registry Integration
- [ ] MLflow tracking for experiment management
- [ ] Model versioning with automatic promotion
- [ ] A/B testing framework for algorithm comparison
- [ ] Rollback capability for production models

### 6. Performance Optimization
- [ ] HNSW approximate nearest neighbors for large datasets (>50k points)
- [ ] Mini-batch DBSCAN for streaming data
- [ ] Feature selection via PCA for dimensionality reduction
- [ ] GPU acceleration for distance computations (optional)

## API Integration

- [ ] `POST /monitors/{id}/anomaly-config` — Configure detection parameters
- [ ] `GET /analytics/anomalies` — Retrieve detected anomalies with context
- [ ] WebSocket: `anomaly.detected` event for real-time streaming
- [ ] Alert rule: `anomaly_score` threshold trigger

## Validation & Monitoring

- [ ] Silhouette score evaluation (> 0.25 target)
- [ ] Custom civic metrics: Detection Rate > 0.85, FPR < 0.10
- [ ] Weekly drift detection (Kolmogorov-Smirnov test)
- [ ] Automated retraining pipeline on drift detection

## Acceptance Criteria

- [ ] DBSCAN correctly identifies density-based clusters on test data
- [ ] Ensemble outperforms any single algorithm by > 10% on F1 score
- [ ] Civic presets achieve < 10% false positive rate on domain test sets
- [ ] Inference latency p99 < 500ms for batch of 100 documents
- [ ] Drift detection triggers correctly on simulated distribution shift
- [ ] All configuration options documented in ML_TUNING.md
- [ ] Unit tests cover edge cases (empty clusters, all noise, single cluster)

## Related

- [ML_TUNING.md](https://github.com/POWDER-RANGER/CIVWATCH/blob/main/ML_TUNING.md)
- [ARCHITECTURE.md — Anomaly Detection Flow](https://github.com/POWDER-RANGER/CIVWATCH/blob/main/ARCHITECTURE.md#anomaly-detection-flow)
- [scikit-learn DBSCAN](https://scikit-learn.org/stable/modules/generated/sklearn.cluster.DBSCAN.html)

---

**Estimated Effort**: 3-4 weeks  
**Priority**: High  
**Target Release**: v0.3.0
